Formula/stroke.rb

```diff
@@ -0,0 +1,17 @@
+class Stroke < Formula
+  desc "🖊️ A MacOS command stroke tool written in Rust 一款基于 Rust 实现的 Mac 命令行一笔画绘线工具"
+  homepage "https://github.com/meloalright/stroke"
+  url "https://github.com/meloalright/stroke/archive/refs/tags/0.1.5.tar.gz"
+  sha256 "cd20d8647272dcedba5eb779e29a7eebe8ec21ec9e2947db893969052bc1aa7a"
+  license "MIT"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    assert_equal "stroke 0.1.5", shell_output("#{bin}/stroke -V").split("\n").last
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
thanks a lot!

3 more function cases have been added in commit [4ed513e](https://github.com/meloalright/homebrew-core/blob/4ed513e1e6e0e9636bdbc127181537fa31f40518/Formula/stroke.rb#L16-L30)